### PR TITLE
Adding method to instantiate from flat json.

### PR DESCRIPTION
**Pull request by THOR300**
This Pull Request: 
---
- Adds a method to the data access library such that we can instantiate a parser output from flat json. 
- This is required as we are pushing flat json to the hugging face repo for visibility reasons. 

We hard code the fields html_data and pdf_data for clear down if they are in the desired state, it may be possible for us to do this dynamically using something like the following to determine if a field would throw an error but this may be rather involved and beyond reasonable scope of this ticket. 

```python 
ParserOutput.model_fields['languages'].annotation
-> typing.Optional[typing.Sequence[str]]
```

Something like the above where we dive into the nested structure to see if None values for a field that come from filling empty fields in hugging face will cause an error.

### CODE REVIEW

File: src/cpr_data_access/utils.py [hunk 1]
```diff
@@ -86,3 +86,29 @@ def dig(obj: Union[list, dict], *fields: Any, default: Any = None) -> Any:
...
+    return unflattened
+
+
+def remove_if_all_nested_vals_none(data: dict, key: str) -> dict:
+    """Remove an item if it's a dict with all values in the nested dictionary are None."""
+    if key not in data:
+        return data
+    if isinstance(data[key], dict):
+        if all(value is None for value in data[key].values()):
+            data.pop(key)
```
Comment (mpjuhasz) [inline src/cpr_data_access/utils.py:113]: Any reason for keeping the field if the value is `None`? More intuitive behaviour for me would be removing. I see that you have a test covering that case, so wondering if that's for any particular reason.

File: src/cpr_data_access/utils.py [hunk 2]
```diff
@@ -86,3 +86,29 @@ def dig(obj: Union[list, dict], *fields: Any, default: Any = None) -> Any:
...
+
+    I.e. metadata.data respresents {"metadata": {"data": {}}}
+    """
+    unflattened = {}
+    for key, value in data.items():
+        parts = key.split(".")
+        current = unflattened
+        for part in parts[:-1]:
+            current = current.setdefault(part, {})
+        current[parts[-1]] = value
```
Comment (mpjuhasz) [inline src/cpr_data_access/utils.py:103]: This is very nice!

File: src/cpr_data_access/utils.py [hunk 3]
```diff
@@ -86,3 +86,29 @@ def dig(obj: Union[list, dict], *fields: Any, default: Any = None) -> Any:
...
+    return unflattened
+
+
+def remove_if_all_nested_vals_none(data: dict, key: str) -> dict:
```
Comment (kdutia) [inline src/cpr_data_access/utils.py:107]: naming suggestion: `remove_key_if_all_nested_vals_none`. I think this then helps you understand what the second arg does when using this method if it's not passed with the `key` keyword

**Review by mpjuhasz — APPROVED** [review]
LGTM -- one minor comment

File: src/cpr_data_access/utils.py [hunk 4]
```diff
@@ -86,3 +86,29 @@ def dig(obj: Union[list, dict], *fields: Any, default: Any = None) -> Any:
...
+    """Remove an item if it's a dict with all values in the nested dictionary are None."""
```
Comment (kdutia) [inline src/cpr_data_access/utils.py:108]: This documentation says to me that `data[key]` can be nested, but this only works if it's flat. Maybe worth updating.

Also, tbh, given this is two lines of readable code used in one place I'd consider not using a separate function here

File: src/cpr_data_access/utils.py (hunk 1 above)
Comment (THOR300) [reply src/cpr_data_access/utils.py:113]: Not sure I follow, in this function we are removing the key if all the values of the the dict are None. 

E.g. {"key": {"a": None, "b": None}} -> {}

**Review by kdutia — APPROVED** [review]

**Pull request merged**